defog/admin_methods.py

```diff
@@ -257,3 +259,133 @@ def get_golden_queries(
         return golden_queries
     else:
         raise ValueError("format must be either 'csv' or 'json'.")
+
+
+def create_table_ddl(
+    table_name: str, columns: List[Dict[str, str]], if_not_exists=True
+) -> str:
+    """
+    Return a DDL statement for creating a table from a list of columns
+    `columns` is a list of dictionaries with the following keys:
+    - column_name: str
+    - data_type: str
+    - column_description: str
+    """
+    md_create = ""
+    if if_not_exists:
+        md_create += f"CREATE TABLE IF NOT EXISTS {table_name} (\n"
+    else:
+        md_create += f"CREATE TABLE {table_name} (\n"
+    for i, column in enumerate(columns):
+        col_name = column["column_name"]
+        # if column name has spaces and hasn't been wrapped in double quotes, wrap it in double quotes
+        if " " in col_name and not col_name.startswith('"'):
+            col_name = f'"{col_name}"'
+        dtype = column["data_type"]
+        if i < len(columns) - 1:
+            md_create += f"  {col_name} {dtype},\n"
+        else:
+            # avoid the trailing comma for the last line
+            md_create += f"  {col_name} {dtype}\n"
+    md_create += ");\n"
+    return md_create
+
+
+def create_ddl_from_metadata(metadata: Dict[str, List[Dict[str, str]]]) -> str:
+    """
+    Return a DDL statement for creating tables from metadata
+    `metadata` is a dictionary with table names as keys and lists of dictionaries as values.
+    Each dictionary in the list has the following keys:
+    - column_name: str
+    - data_type: str
+    - column_description: str
+    """
+    md_create = ""
+    for table_name, columns in metadata.items():
+        if "." in table_name:
+            table_name = table_name.split(".")[-1]
+            schema_name = table_name.split(".")[0]
+
+            md_create += f"CREATE SCHEMA IF NOT EXISTS {schema_name};\n"
+        md_create += create_table_ddl(table_name, columns)
+    return md_create
+
+
+def create_empty_tables(self, dev: bool = False):
+    """
+    Create empty tables based on metadata
+    """
+    metadata = self.get_metadata(format="json", dev=dev)
+    if self.db_type == "sqlserver":
+        ddl = create_ddl_from_metadata(metadata, if_not_exists=False)
+    else:
+        ddl = create_ddl_from_metadata(metadata)
+
+    if self.db_type == "postgres" or self.db_type == "redshift":
```

Comment:
nit: do we want to wrap the code from here to the end below in a try/except block in case the metadata is malformed and can't be run? We can also just bubble common things like `ImportError` through.